### PR TITLE
feat: Improve zod error messages

### PR DESCRIPTION
**Pull request by sradevski**

### CODE REVIEW

**Comment by olivermrbl** [comment]
This is really great, how much can we customize the error messages? Think the `at` reads a little bit weird

**Comment by sradevski** [comment]
@olivermrbl I was just checking that, I think eventually we might want to roll out our own formatter, esp. when it comes to localization, but I'll try and see if we can at least make this a bit more user-friendly

**Comment by sradevski** [comment]
@olivermrbl ignore what you saw, check it again 😄 At the end it seems just implementing it manually is better.

File: packages/medusa/src/api/utils/__tests__/zod-helper.spec.ts [hunk 1]
```diff
@@ -0,0 +1,134 @@
+import { z } from "zod"
```
Comment (sradevski) [inline packages/medusa/src/api/utils/__tests__/zod-helper.spec.ts:1]: I renamed validate-body to zod-helper tests, I didn't really delete anything 😄

File: packages/medusa/src/api/utils/zod-helper.ts [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+const formatUnionError = (issue: ZodInvalidUnionIssue) => {
+  const issues = issue.unionErrors.flatMap((e) => e.issues)
+  return (
+    formatInvalidType(issues) || formatRequiredField(issues) || issue.message
+  )
+}
+
+const formatError = (err: ZodError) => {
+  const issueMessages = err.issues.slice(0, 3).map((issue) => {
+    console.log(issue)
```
Comment (olivermrbl) [inline packages/medusa/src/api/utils/zod-helper.ts]:
```suggestion
```

**Review by olivermrbl — APPROVED** [review]
Very nice @sradevski !

**Pull request merged**